java/shared/java/org/signal/libsignal/internal/Native.java

```diff
@@ -49,8 +49,14 @@ private static void copyToTempFileAndLoad(InputStream in, String extension) thro
   private static void loadLibrary() {
     try {
       String  osName    = System.getProperty("os.name").toLowerCase(java.util.Locale.ROOT);
-      boolean isMacOs   = osName.startsWith("mac os");
-      String  extension = isMacOs ? ".dylib" : ".so";
+      String extension;
+      if (osName.startsWith("mac os")) {
+        extension = ".dylib";
+      } else if(osName.contains("win")) {
```

Comment:
Does `startsWith` not work? Seems good to keep the check narrow.

Reply:
I think the startsWIth should work indeed